network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py

```diff
@@ -341,6 +342,18 @@ def generate_topic_entries_set_html(self, request, topic, *args, **kwargs):
 
         return self.generate_entries_set_html(request, *args, **kwargs)
 
+    @route(r"^category/(?P<topic_slug>.+)/")
+    def category_to_topic_redirect_view(self, request, topic_slug):
+        """
+        If a user attempts to visit the "entries by topic" page using
+        the deprecated /category/ route, we want to redirect them to the
+        correct page using the /topic/ route.
+        """
+        blog_index_slug = self.slug
+        return localized_redirect(
+            request, destination_path=f"{blog_index_slug}/topic", subpath=topic_slug, is_permanent=True
+        )
```

Comment:
I am surprised by this utils function (I have noticed we have a couple of those around and am wondering about their use). Have you consider generating the redirect url using `self.reverse_subpage` to get the topic URL? The necessary information about the locale etc. should already be on the page itself. So I am not sure for the need of that custom utils function. 

See also: https://docs.wagtail.org/en/stable/reference/contrib/routablepage.html#wagtail.contrib.routable_page.models.RoutablePageMixin.reverse_subpage

Reply:
Hi @tbrlpld thanks for the great feedback regarding the redirect.

After looking into it a little more, I tried to see if I can rework this function to instead use django's `redirect()` function instead of our `localized_redirect`, but found it quite difficult to get both the language code **and** query parameters to work.

For that reason, I am voting that we stick with this custom `localized_redirect` function.

However, I do like the idea of using `self.reverse_subpage`, as that means we wont have the path `/topic/` hardcoded. I have updated the function to use reverse subpage to grab the needed path:

https://github.com/MozillaFoundation/foundation.mozilla.org/blob/4927fe1e045b6f79e9298e85f9fe256e47785f8e/network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py#L352-L355

Reply:
Sorry for being inquisitive about this. I have just been noticing these localization utils all over the place and am not sure why they exist. Understanding their purpose is especially hard since there are no comments and no tests for them. If find them quite confusing because their functionality seems to be duplicating things that come out-of-the-box or simpler with Wagtail.
 
Can you add some more context of what is going wrong when trying to combine queryparameter and language code? What do you have to do to make the language code work? Is is exactly the part that I meant should come from using `self.reverse_subpage`. The queryparameters you might have to tag on. 

If the point of using `localized_redirect` is to keep queryparameters, we might be better off with a util for that specific use case. And probably also better to make that util only add the queryparameters from a request to a URL string (instead of tying it to the redirect functionality) because that increased the use cases for it.

Reply:
Hi @tbrlpld!

After taking a second look at the docs I believe I found my mistake. 🤦 

I was attempting to redirect to the path returned by `self.reverse_subpage` like so:

```python
topic_path = self.reverse_subpage("entries_by_topic", args=(topic_slug,))

...
topic_path += f'?{request.META["QUERY_STRING"]}'

return redirect(topic_path, permanent=True)
```

Doing so would create some sort of infinite loop and redirect me to a path such as:
`localhost:8000/blog/topic/mozilla-festival/topic/mozilla-festival/topic/mozilla-festival/topic/mozilla-festival/topic/mozilla-festival/...`



However, after taking a 2nd look at the docs I see that they combine both `self.url` and `self.reverse_subpage` like so:
```python
url = page.url + page.reverse_subpage('events_for_year', kwargs={'year': '2014'})
```


After updating my code to the following, everything is working as expected (language code, query params, etc):
```python
topic_path = self.url + self.reverse_subpage("entries_by_topic", args=(topic_slug,))

if request.META["QUERY_STRING"]:
  topic_path += f'?{request.META["QUERY_STRING"]}'

return redirect(topic_path, permanent=True)
```